C-Plus-Plus/cp/tower_of_hanoi.cpp

```diff
@@ -0,0 +1,46 @@
+/* Tower Of Hanoi
+Tower of Hanoi is a mathematical puzzle where we have three rods and n disks. The objective of the puzzle is to move the entire stack to another rod using the below rules:
+*Only one disk can be moved at a time.
+*Each move consists of taking the upper disk from one of the stacks and placing it on top of another stack i.e. a disk can only be moved if it is the uppermost disk on a stack.
+*No disk must be placed on top of a smaller disk.
+*/
+#include <bits/stdc++.h>
+using namespace std;
+void towerOfHanoi(int n, char from_rod, char to_rod, char intermediate_rod)
+{
+    if (n == 1)
+    {
+        cout << "Move disk 1 from rod " << from_rod << " to rod " << to_rod << endl;
+        return;
+    }
+    towerOfHanoi(n - 1, from_rod, intermediate_rod, to_rod);
+    cout << "Move disk " << n << " from rod " << from_rod << " to rod " << to_rod << endl;
+    towerOfHanoi(n - 1, intermediate_rod, to_rod, from_rod);
+}
+int main()
+{
+    int n;
+    cin >> n;
+    towerOfHanoi(n, 'A', 'C', 'B');
+}
+/*
+Time Complexity: O(n)
```

Comment:
Add Time and Space Complexity at the end of the Code   
Also Time complexity of Tower of Hanoi problem using recursion is O( 2^n - 1)  
How it is O(n)?

Reply:
@ankitaggarwal23 Sorry for not being able to commit the changes earlier. I have created a new pull request https://github.com/TesseractCoding/NeoAlgo/pull/3980/